lib/chef/provider/user/linux.rb

```diff
@@ -37,6 +37,8 @@ def compare_user
               @change_desc << "change #{user_attrib} from #{cur_val} to #{new_val}"
             end
           end
+
+          !@change_desc.empty?
```

Comment:
This is a little opaque - this function relies on the fact that super also modifies `@change_desc`. That's internal to the base class though, and not something that we should rely on.

Could we instead do something like: 
```
user_changed  = super
...
-snip-
...
user_changed || !change_desc.empty? 
```

Reply:
Hi Marc, good input, I changed it.

Reply:
In this case @change_desc should be initialized after the call to super too or the risk is that super didn't create the change_desc property and ` @change_desc << "change #{user_attrib} from #{cur_val} to #{new_val}"` will raise a no method error for << on nil.

Reply:
Another approach could be as in the mac user provider to rewrite the whole method: https://github.com/chef/chef/blob/main/lib/chef/provider/user/mac.rb#L222-L234